vermouth/processors/do_mapping.py

```diff
@@ -323,12 +329,20 @@ def do_mapping(molecule, mappings, to_ff, attribute_keep=()):
                 if out_idx != out_jdx:
                     graph_out.add_edge(out_idx, out_jdx)
         if shared_atoms:
-            print("You have a shared atom between blocks. This may mean you"
-                  " have too  particles in your output and/or erroneous bonds.")
+            LOGGER.warning("You have the following atoms that are shared"
+                           " between blocks. This may mean you have too many"
+                           " particles in your output and/or erroneous bonds."
+                           " {}. They've end up in the following output atoms:"
+                           " {}.",
+                           [molecule.nodes[idx] for idx in shared_atoms],
+                           [graph_out.nodes[idx] for idx in shared_out_atoms])
     # TODO: These should be turned into warnings.
-    print('double covered:', {k: len(v) for k, v in mol_to_out.items() if len(v) > 1})
-    print('uncovered:', set(molecule.nodes.keys()) - set(mol_to_out.keys()))
-    print(len(set(molecule.nodes.keys()) - set(mol_to_out.keys())))
+    LOGGER.debug('double covered: {}', {k: v for k, v in blocks_per_atom.items() if v > 1})
+    for node_idx, v in blocks_per_atom.items():
+        if v > 1:
+            LOGGER.debug('{}', molecule.nodes[node_idx]['atomname'])
```

Comment:
Debug messages should ideally be standalone. This loop of messages would not be.